Infrastructure-Scripts/Get-EnvironmentDeploymentRecords.ps1

```diff
@@ -0,0 +1,62 @@
+<#
+    .SYNOPSIS
+    Gets the latest deployment record for a given pipeline and environment and checks whether the current instance is able to run without clashing with said record.
+
+    .DESCRIPTION
+    Gets payload from Azure DevOps RestAPI and filters it to produce the build ID of the newest running build. Then compares to its own build ID, if equal to it then run and if not sleep and redo.
+
+    .PARAMETER EnvironmentId
+    The ID number associated with the environemnt stage being used for the deployment,
+
+    .PARAMETER PipelineName
+    The name of the pipeline that the deployment is being run on.
+
+    .PARAMETER RunId
+    The Run ID of the current deployment which is used to compare to the lowest active RunID returned by the API
+
+    .PARAMETER SleepTime
+    The the amount of time in seconds that the process will wait before retrying the comparison if there is another deployment currently running.
+
+    .EXAMPLE
+    .\Get-EnvironmentDeploymentRecords.ps1 -EnvironmentId 139 -PipelineName das-levy-transfer-matching-api -RunId 460XXX
+#>
+
+[CmdletBinding()]
+Param (
+    [Parameter(Mandatory = $true)]
+    [String]$EnvironmentId,
+    [Parameter(Mandatory = $true)]
+    [String]$PipelineName,
+    [Parameter(Mandatory = $true)]
+    [Int]$RunId,
+    [Parameter(Mandatory = $false)]
+    [Int]$SleepTime = 20
+)
+
+$Url = "$env:SYSTEM_ORGANISATIONNAME/$env:SYSTEM_PROJECTNAME/_apis/distributedtask/environments/$EnvironmentId/environmentdeploymentrecords?top=100?api-version=6.0-preview.1"
+$RetryCounter = 1
+
+while ($RetryCounter -lt 10){
```

Comment:
If the default value for SleepTime is used the loop will run for just over 3 minutes, that seems a bit short as a typical deployment takes 3 - 5 minutes.  I'd say it should run for at least 10 minutes (double a typical deployment) before timing out

Reply:
While condition is now set to 30 iterations